enumerable.rb

```diff
@@ -0,0 +1,146 @@
+# rubocop: disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    i = 0
+    array = []
+    while i < size
+      yield to_a[i]
+      array << to_a[i]
+      i += 1
+    end
+    array
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    i = 0
+    array = []
+    while i < size
+      yield to_a[i], i
+      array << to_a[i]
+      i += 1
+    end
+    array
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    my_each do |x|
+      array.push(x) if yield(x) == true
+    end
+    array
+  end
+
+  def my_all?(reg = nil)
+    if !reg.nil?
+      my_each { |x| return false unless reg == x }
+    elsif block_given?
+      my_each { |x| return false unless yield x }
+    else
+      my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?(reg = nil)
+    res = false
+    if !reg.nil?
+      my_each { |x| res = true if reg == x }
+    elsif block_given?
+      my_each { |x| res = true if yield(x) }
+    else
+      my_each { |x| res = true if x }
+    end
+    res
+  end
+
+  def my_none?(_proc = nil)
+    res = false
+    if !proc.nil?
+      my_each { |x| res = true if reg == x }
+    elsif block_given?
+      my_each { |x| res = true if yield(x) }
```

Comment:
It should say the opposite. my_none should return false if one of the elements in the enumerable is true.